lib/SILOptimizer/IPO/LetPropertiesOpts.cpp

```diff
@@ -529,17 +485,38 @@ void LetPropertiesOpt::collectStructPropertiesAccess(StructInst *SI,
     if (SkipProcessing.count(Prop))
       continue;
     SILValue PropValue = SI->getOperandForField(Prop)->get();
-    DEBUG(llvm::dbgs() << "Check the value of property '" << Prop->getName()
+    DEBUG(llvm::dbgs() << "Check the value of property '" << *Prop
                        << "' :" << PropValue << "\n");
     if (!analyzeInitValue(SI, Prop)) {
       SkipProcessing.insert(Prop);
-      DEBUG(llvm::dbgs() << "The value of a let property '"
-                         << structDecl->getName() << "::" << Prop->getName()
+      DEBUG(llvm::dbgs() << "The value of a let propertiy '" << *Prop
                          << "' is not statically known\n");
     }
   }
 }
 
+/// Check if I is a sequence of projections followed by a load.
+/// Since it is supposed to be a load from a let property with
+/// statically known constant intializer, only struct_element_addr
+/// and tuple_element_addr projections are considered.
+static bool isValidPropertyLoad(SILInstruction *I) {
+  if (isa<LoadInst>(I))
+    return true;
+
+  if (auto *SEAI = dyn_cast<StructElementAddrInst>(I)) {
+    if (SEAI->hasOneUse())
```

Comment:
Restricting to a single use seems overly conservative. Why not call isValidPropertyLoad for all uses?